### PR TITLE
Add stat, lstat

### PR DESCRIPTION
**Pull request by mitchellwrosen**
#38

### CODE REVIEW

File: src/Turtle/Prelude.hs [hunk 1]
```diff
@@ -228,6 +228,29 @@ module Turtle.Prelude (
...
+    , PosixCompat.deviceID
+    , PosixCompat.fileID
+    , PosixCompat.fileMode
+    , PosixCompat.linkCount
+    , PosixCompat.fileOwner
+    , PosixCompat.fileGroup
+    , PosixCompat.specialDeviceID
+    , PosixCompat.fileSize
+    , PosixCompat.accessTime
+    , PosixCompat.modificationTime
```
Comment (Gabriel439) [inline src/Turtle/Prelude.hs]: For `stat`-related functions that return an `EpochTime` like `accessTime` and `modificationTime`, could we wrap them to return a `time` library type like `PosixTime`?

File: src/Turtle/Prelude.hs [hunk 2]
```diff
@@ -228,6 +228,29 @@ module Turtle.Prelude (
...
+    , stat
+    , lstat
+    , PosixCompat.deviceID
+    , PosixCompat.fileID
+    , PosixCompat.fileMode
+    , PosixCompat.linkCount
+    , PosixCompat.fileOwner
+    , PosixCompat.fileGroup
+    , PosixCompat.specialDeviceID
+    , PosixCompat.fileSize
```
Comment (Gabriel439) [inline src/Turtle/Prelude.hs]: Could we wrap this to return a `Size`?

File: src/Turtle/Prelude.hs [hunk 3]
```diff
@@ -228,6 +228,29 @@ module Turtle.Prelude (
...
+    , PosixCompat.FileStatus
+    , stat
+    , lstat
+    , PosixCompat.deviceID
+    , PosixCompat.fileID
+    , PosixCompat.fileMode
+    , PosixCompat.linkCount
+    , PosixCompat.fileOwner
+    , PosixCompat.fileGroup
+    , PosixCompat.specialDeviceID
```
Comment (Gabriel439) [inline src/Turtle/Prelude.hs]: For all the functions from `specialDeviceID` and up I'd suggest removing them until somebody needs them.  My reasoning is that it's not clear what more opinionated and idiomatic type to wrap them in, and we won't know the answer to that until somebody needs to use them.

Comment (Gabriel439) [inline src/Turtle/Prelude.hs]: ... except of course the `stat` and `lstat` functions.  Keep those! :)

**Comment by Gabriel439** [comment]
Other than those comments this looks good to me

**Comment by mitchellwrosen** [comment]
Done and force-pushed

**Comment by Gabriel439** [comment]
Awesome!  Thank you :)

Also, for future reference, you don't need to force push if you want a clean history.  Github provides me with the option to perform a squash merge that generates a single commit from your branch, no matter how messy it is.

**Comment by mitchellwrosen** [comment]
Oh, I didn't know that, thanks!

**Pull request merged**